### DISCUSSION TITLE
Welcome to merelinux Discussions!

### DISCUSSION THREAD

**Discussion by jhuntwork** [jhuntwork/merelinux#58] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jhuntwork** [comment]
Hello, I'm @jhuntwork, the maintainer of this repository. I enabled the Discussions feature here to have an easy place to collaborate and share ideas. Originally, I wanted a mailing list for interested parties to subscribe to, but this may be an easier option as we're just starting out. Feel free to ask any questions or offer any ideas here.

**Comment by fungilife** [comment]
A friend let me know that you are active again, I need to take my mere installation out of the freezer and update.
Last efforts was to try to build simple packages and run into libraries that couldn't coexist.  I'll have to look at my notes.  I wouldn't mind trying your testing repository as well, but I seem to only know of [main]

musl + s6 + pacman    would have been my choices to start at a distro of my own as well, I just accept I haven't matured enough to try it.   Even with glibc, I tried to build one by one the very basic Arch core pkgs and then move base-devel bunch, but in absence of systemd/-libs,elogind .... and I was surprised that it fails.  I know that base includes those, I meant base without them.  Some pkgs fail on checks of interdependent pkgs, and one can choose to skip the tests and hope the functionality is maintained....  but Arch (and arch based) became hopeless.

I liked void and I could build xbps in arch but couldn't build pacman in void/musl.  So mere for me is 3 steps in the right direction.

**Reply by jhuntwork** [reply]
Nice! Thanks for your comments @fungilife. I'm glad Mere is heading in the right direction.

If you want to play with the testing repo, you can just change the `Server` setting to use the testing path in pacman.conf, like:

```
[main]
Server = http://pkgs.merelinux.org/testing
```

**Comment by worldpe** [comment]
What's the reason for using pacman?

**Reply by jhuntwork** [reply]
When I started, It was a lightweight, complete package manager with few dependencies and little fuss. I liked the way it described its build steps as simple bash functions (by comparison to rpm specs which I had used before and were just overly complex and confusing).

Using pacman let me get started and focus on the toolchain and other distribution questions without getting too bogged down with other issues.

There are features and concepts that I'd like to explore via a different package manager at some point, but for now I'm happy to put those to the side as we focus on getting a stable system.

**Comment by 4L1V3iC** [comment]
Hello there. I am very interested in learning more about and possibly contributing towards the success of your project. Please send me an email to establish contact: 4L1V3@protonmail.com

**Reply by subnut** [reply]
_joined 4 hours ago_

🤨

**Comment by fred-sheehan** [comment]
Good to see another OS developer looking to build on good foundations, I shall keep an eye on this one.

**Reply by jhuntwork** [reply]
Thanks for dropping a line! Always looking for feedback too, so please feel free to share. I'm working on having some install instructions soon so people can begin testing and even helping develop more features.

**Comment by GroosL** [comment]
Great distro! Always good to see diferent projects like this. 

Is there any reason to not use Obarun's 66? It's s6 but easier.

**Reply by mchalunderscore** [reply]
1-up on using 66, iirc it's a pretty compatible drop-in replacement

**Reply by jhuntwork** [reply]
I haven't really tried 66. What are the advantages, how is it easier? I could be wrong, but I think when I started using s6, 66 wasn't around so wrote a little shell utility to manage service state around it: https://github.com/jhuntwork/merelinux/blob/main/packages/s6/s6-service

I don't mind taking a look at 66, but I do like keeping dependencies light and implementations simple, so there would need to be some pretty obvious advantages. Care to fill me in with what you see those as being?

**Reply by mchalunderscore** [reply]
66 is built up on top of s6 and s6-rc, keeping the same dependencies but generally offering a larger "suite" of utilities (hence its full name, suite66)
it's also pretty simple to get it running on other systems, pretty much just as easy as setting up the right config skel in /etc, dropping the binaries into the right places and pointing the kernel at 66-boot for init
the main HUGE advantage in my eyes however is the service management system, it vastly simplifies the trees system s6/s6-rc already has and offers simple commands for managing them (e.g. 66-enable -t \<tree> \<servicename>), while also keeping a fairly sane service frontend file format (which you can see at https://web.obarun.org/software/66/latest/frontend.html)
the obarun folks who create and manage 66 also of course use pacman as their package manager, so you could even use their packages as examples for packaging 66 on mere

**Reply by jhuntwork** [reply]
Nice, thanks!

Well, pull requests are welcome. I'd be happy to review something that brings it in. It would also probably shine a light on any gaps in documentation or process that exist for development.

**Reply by mchalunderscore** [reply]
i'd be happy to maybe open a pr and try to do a basic implementation of 66 over the next week or so!

**Reply by jhuntwork** [reply]
@jnats had a chance to look at this yet? No pressure, just want to make sure you have the info you need for getting started. Let me know if you have any questions, please.

**Comment by fungilife** [comment]
Before 66 there was s6opts, but I have used 66 since its alpha phase.  There is and has never been any reason in the past 3 years that I would use s6 without 66.  To me that is madness.  It is like refusing to use a throttle and clutch on a motorcycle and rather reach into the motor while riding to move the slide up and down and pull the clutch basket out by hand, while riding!  That much of a difference.

The suite is using skalibs and execline, it adds its own complimentary library to skalibs, called oblibs, and 66 for core admin tools, 66-tools for additional sys-admin hacking (not absolutely necessary for 66).  
There are service files (sensitive to distro and services themselves) service modules, and module-type services.  https://wiki.obarun.org/doku.php?id=66intro   shows how 66 works in practice.

It is all C, and execline.  It has worked pretty much anywhere it was applied, it takes a few minutes to build.
Void (glibc & musl) (64 & 32bit) (all architectures)
Antix, devuan, alpine, adelie....even gobo!    Officially in Artix, in Void you can use the mobinmob repository to install it besides or in place of runit.  https://codeberg.org/mobinmob/

Any deeper technical questions you may have talk to the author directly eric@obarun.org

You can also download an image of Obarun and boot it, use the intro to 66 as a guide, and play around with the tools.  % pacman -Ql 66  pacman -Ql 66-tools  shows you the contents.  (man html and md documentation is included).
